test/test_case.rb

```diff
@@ -0,0 +1,41 @@
+# typed: true
+# frozen_string_literal: true
+
+# NOTE: This module is intended to be used by addons for writing their own tests, so keep that in mind if changing.
+
+module RubyLsp
+  module TestCase
+    extend T::Sig
+
+    sig do
```

Comment:
Should we remove all the Sorbet aspects from this, so that we're not forcing addon authors to use it?